index.js

```diff
@@ -302,7 +302,24 @@ const createInstance = (defaults = {}) => {
 	const ky = (input, options) => new Ky(input, deepMerge({}, defaults, options));
 
 	for (const method of requestMethods) {
-		ky[method] = (input, options) => new Ky(input, deepMerge({}, defaults, options, {method}));
+		ky[method] = (input, options = {}) => {
+			const {headers: providedHeaders, ...restOfOptions} = options;
+			const headers = new Headers(providedHeaders);
+			if (!headers.has('Accept')) {
+				headers.set('Accept', 'application/json');
```

Comment:
This is inappropriate - it should be set only if someone does `ky(...).json()`.

Reply:
@szmarczak I've explicitly asked about this [here](https://github.com/sindresorhus/ky/issues/83#issuecomment-478900902) and oyu confirmed it. 
Besides, i don't believe it's technically possible to set accept headers once `json()` is called on the return value from `ky(...)` - the request has already been made right?

Reply:
I have missed this part:

> there needs to be Accept: application/json header set by default.

Anyway as you can read from the OP:

> One thing that got does, which I think it awesome, and which we don't have in ky, is setting the accept header when .json() is called.

So yeah, we need to iterate over every shortcut method and just replace `.json()` I think.